schema/pod.go

```diff
@@ -116,6 +116,11 @@ func (al AppList) assertValid() error {
 			return fmt.Errorf(`duplicate apps of name %q`, a.Name)
 		}
 		seen[a.Name] = true
+		if a.App != nil {
```

Comment:
This links `AppList` validation to validation of `RuntimeApp` instances contained therein.

But I think I see your point now: I was considering this validation function per-se, while in the bigger context it already makes sense as every layer performs its own validation. Given that this is a private validation method only uses by the marshaller, I'll remove this un-necessary snippet.